#!/bin/bash

#conda activate Homer
cd /Seq_analysistools/anaconda3/envs/Homer/share/homer-4.10-0

echo "KW: starting counting reads across genes for MANE genes, normalizing to 100 million total mapped reads"
echo "KW: filtering out genes without at least 1-fold enrichment in the gene vs 1kb directly upstream (RPKM)"
echo "KW: using GTF file with only isolated expressed MANE genes (UNstranded iso filter, 5,576 genes)"
date
echo "KW: starting with unstranded isolated genes, counting reads only on same strand as gene annotation"
#not sure if this normalizes counts by length of region counted, otherwise the length of genes and length of upstream window matter a lot. Using -RPKM to know it's norm for length
analyzeRepeats.pl /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/06_defining_gene_lists/custom_MANE_GTF_files/iso_exp_unstrand_MANE_v095_from_table_browser.gtf hg38 -count genes -strand + -rpkm -norm 1e8 -upstream -2000 -L 1 -d /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/03_TDs/rep1_HS_hg38_TD > /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/06_defining_gene_lists/defining_read_in/unstranded_starting_iso_exp_genes_gtf/stranded_upstream_2kb_1fold/rep1_HS_hg38.txt
analyzeRepeats.pl /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/06_defining_gene_lists/custom_MANE_GTF_files/iso_exp_unstrand_MANE_v095_from_table_browser.gtf hg38 -count genes -strand + -rpkm -norm 1e8 -upstream -2000 -L 1 -d /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/03_TDs/rep1_NHS_hg38_TD > /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/06_defining_gene_lists/defining_read_in/unstranded_starting_iso_exp_genes_gtf/stranded_upstream_2kb_1fold/rep1_NHS_hg38.txt
analyzeRepeats.pl /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/06_defining_gene_lists/custom_MANE_GTF_files/iso_exp_unstrand_MANE_v095_from_table_browser.gtf hg38 -count genes -strand + -rpkm -norm 1e8 -upstream -2000 -L 1 -d /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/03_TDs/rep2_HS_hg38_TD > /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/06_defining_gene_lists/defining_read_in/unstranded_starting_iso_exp_genes_gtf/stranded_upstream_2kb_1fold/rep2_HS_hg38.txt
analyzeRepeats.pl /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/06_defining_gene_lists/custom_MANE_GTF_files/iso_exp_unstrand_MANE_v095_from_table_browser.gtf hg38 -count genes -strand + -rpkm -norm 1e8 -upstream -2000 -L 1 -d /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/03_TDs/rep2_NHS_hg38_TD > /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/06_defining_gene_lists/defining_read_in/unstranded_starting_iso_exp_genes_gtf/stranded_upstream_2kb_1fold/rep2_NHS_hg38.txt
analyzeRepeats.pl /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/06_defining_gene_lists/custom_MANE_GTF_files/iso_exp_unstrand_MANE_v095_from_table_browser.gtf hg38 -count genes -strand + -rpkm -norm 1e8 -upstream -2000 -L 1 -d /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/03_TDs/rep3_HS_hg38_TD > /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/06_defining_gene_lists/defining_read_in/unstranded_starting_iso_exp_genes_gtf/stranded_upstream_2kb_1fold/rep3_HS_hg38.txt
analyzeRepeats.pl /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/06_defining_gene_lists/custom_MANE_GTF_files/iso_exp_unstrand_MANE_v095_from_table_browser.gtf hg38 -count genes -strand + -rpkm -norm 1e8 -upstream -2000 -L 1 -d /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/03_TDs/rep3_NHS_hg38_TD > /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/06_defining_gene_lists/defining_read_in/unstranded_starting_iso_exp_genes_gtf/stranded_upstream_2kb_1fold/rep3_NHS_hg38.txt
#strand + means count reads on the same strand as gene annotation, default is unstranded (below)

## I don't think it's fair to use this unstranded because there's no way to skip the divergent txn so lots of genes get filtered out even though there isn't true read-in
#date
#echo "KW: starting stranded counts"
#analyzeRepeats.pl /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/06_defining_gene_lists/custom_MANE_GTF_files/iso_exp_strand_MANE_v095_from_table_browser.gtf hg38 -count genes -norm 1e8 -upstream -2000 -L 5 -d /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/03_TDs/rep1_HS_hg38_TD > /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/06_defining_gene_lists/defining_read_in/unstranded/rep1_HS_hg38.txt
#analyzeRepeats.pl /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/06_defining_gene_lists/custom_MANE_GTF_files/iso_exp_strand_MANE_v095_from_table_browser.gtf hg38 -count genes -norm 1e8 -upstream -2000 -L 5 -d /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/03_TDs/rep1_NHS_hg38_TD > /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/06_defining_gene_lists/defining_read_in/unstranded/rep1_NHS_hg38.txt
#analyzeRepeats.pl /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/06_defining_gene_lists/custom_MANE_GTF_files/iso_exp_strand_MANE_v095_from_table_browser.gtf hg38 -count genes -norm 1e8 -upstream -2000 -L 5 -d /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/03_TDs/rep2_HS_hg38_TD > /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/06_defining_gene_lists/defining_read_in/unstranded/rep2_HS_hg38.txt
#analyzeRepeats.pl /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/06_defining_gene_lists/custom_MANE_GTF_files/iso_exp_strand_MANE_v095_from_table_browser.gtf hg38 -count genes -norm 1e8 -upstream -2000 -L 5 -d /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/03_TDs/rep2_NHS_hg38_TD > /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/06_defining_gene_lists/defining_read_in/unstranded/rep2_NHS_hg38.txt
#analyzeRepeats.pl /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/06_defining_gene_lists/custom_MANE_GTF_files/iso_exp_strand_MANE_v095_from_table_browser.gtf hg38 -count genes -norm 1e8 -upstream -2000 -L 5 -d /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/03_TDs/rep3_HS_hg38_TD > /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/06_defining_gene_lists/defining_read_in/unstranded/rep3_HS_hg38.txt
#analyzeRepeats.pl /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/06_defining_gene_lists/custom_MANE_GTF_files/iso_exp_strand_MANE_v095_from_table_browser.gtf hg38 -count genes -norm 1e8 -upstream -2000 -L 5 -d /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/03_TDs/rep3_NHS_hg38_TD > /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/06_defining_gene_lists/defining_read_in/unstranded/rep3_NHS_hg38.txt


date
echo "KW: using GTF file with only isolated expressed MANE genes (stranded iso filter, 8,447 genes)"
echo "KW: now for stranded isolated genes, counting reads only on same strand as gene annotation"
analyzeRepeats.pl /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/06_defining_gene_lists/custom_MANE_GTF_files/iso_exp_strand_MANE_v095_from_table_browser.gtf hg38 -count genes -strand + -rpkm -norm 1e8 -upstream -2000 -L 1 -d /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/03_TDs/rep1_HS_hg38_TD > /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/06_defining_gene_lists/defining_read_in/stranded_iso_exp_genes_starting_gtf/stranded_upstream_2kb_1fold/rep1_HS_hg38.txt
analyzeRepeats.pl /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/06_defining_gene_lists/custom_MANE_GTF_files/iso_exp_strand_MANE_v095_from_table_browser.gtf hg38 -count genes -strand + -rpkm -norm 1e8 -upstream -2000 -L 1 -d /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/03_TDs/rep1_NHS_hg38_TD > /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/06_defining_gene_lists/defining_read_in/stranded_iso_exp_genes_starting_gtf/stranded_upstream_2kb_1fold/rep1_NHS_hg38.txt
analyzeRepeats.pl /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/06_defining_gene_lists/custom_MANE_GTF_files/iso_exp_strand_MANE_v095_from_table_browser.gtf hg38 -count genes -strand + -rpkm -norm 1e8 -upstream -2000 -L 1 -d /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/03_TDs/rep2_HS_hg38_TD > /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/06_defining_gene_lists/defining_read_in/stranded_iso_exp_genes_starting_gtf/stranded_upstream_2kb_1fold/rep2_HS_hg38.txt
analyzeRepeats.pl /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/06_defining_gene_lists/custom_MANE_GTF_files/iso_exp_strand_MANE_v095_from_table_browser.gtf hg38 -count genes -strand + -rpkm -norm 1e8 -upstream -2000 -L 1 -d /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/03_TDs/rep2_NHS_hg38_TD > /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/06_defining_gene_lists/defining_read_in/stranded_iso_exp_genes_starting_gtf/stranded_upstream_2kb_1fold/rep2_NHS_hg38.txt
analyzeRepeats.pl /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/06_defining_gene_lists/custom_MANE_GTF_files/iso_exp_strand_MANE_v095_from_table_browser.gtf hg38 -count genes -strand + -rpkm -norm 1e8 -upstream -2000 -L 1 -d /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/03_TDs/rep3_HS_hg38_TD > /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/06_defining_gene_lists/defining_read_in/stranded_iso_exp_genes_starting_gtf/stranded_upstream_2kb_1fold/rep3_HS_hg38.txt
analyzeRepeats.pl /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/06_defining_gene_lists/custom_MANE_GTF_files/iso_exp_strand_MANE_v095_from_table_browser.gtf hg38 -count genes -strand + -rpkm -norm 1e8 -upstream -2000 -L 1 -d /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/03_TDs/rep3_NHS_hg38_TD > /Users/goodrichlab/Desktop/Katy/HS_4sU-Seq_2021/06_defining_gene_lists/defining_read_in/stranded_iso_exp_genes_starting_gtf/stranded_upstream_2kb_1fold/rep3_NHS_hg38.txt


date
echo "ALL DONE"